tools/wall_maker.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+"""Wall maker"""
+# FIXME: missing purpose of module in doc
+
+import os
+import sys
+import tempfile
+
+import argparse
+import requests
+from PIL import Image
+
+sys.path.append("pixray")
+import pixray  # noqa pylint: disable=C0413
+
+parser = argparse.ArgumentParser()
+# FIXME: missing argument descriptions
+parser.add_argument(
+    'name', type=str,
+    help=""
+)
+parser.add_argument(
+    'palette', type=str,
+    help=""
+)
+parser.add_argument(
+    'prompt', type=str,
+    help=""
+)
+
+
+def generate_horizontal(name, prompt, palette, overlay_url):
+    """Generate horizontal"""
+    # FIXME: missing meaningful doc
+
+    overlay_temp_filename = os.path.join(tempfile.gettempdir(), 'overlay.png')
+    # download overlay
+    img_data = requests.get(overlay_url).content
+    with open(overlay_temp_filename, 'wb') as handler:
+        handler.write(img_data)
```

Comment:
Overlay needs to be downloaded to be processed.

Using tempfile might be overkill and/or could be done better.